### PR TITLE
fix: fetch response parsing, create safe retry

### PR DESCRIPTION
**Pull request by truemiller**
## Proposed changes

- update API services to parse error statuses and throw
- add utf-8 charset, unlikely to be required, but may support decoding issues on middleware 
- add create safe retries
- updates "creating safe" text at different steps
- check backup signer exists before continuing to main (stops missing backup wallet alert appearing)
- added error message notifications during creation (i.e. if retrying)

no real UI changes, will ask @iasonrovis to test rc162

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: frontend/components/SetupPage/Create/SetupCreateSafe.tsx [hunk 1]
```diff
@@ -1,40 +1,100 @@
...
+          }
+
+          // Set states for successful creation
+          setIsCreatingSafe(false);
+          setIsCreateSafeSuccessful(true);
+          setFailed(false);
+        })
+        .catch(async (e) => {
+          console.error(e);
+          // Wait for 1 second before retrying
```
Comment (truemiller) [inline frontend/components/SetupPage/Create/SetupCreateSafe.tsx:59]:
```suggestion
          // Wait for 5 seconds before retrying
```

File: frontend/components/SetupPage/Create/SetupCreateSafe.tsx [hunk 2]
```diff
@@ -1,40 +1,100 @@
...
+          // Wait for 5 seconds before retrying
+          await new Promise((resolve) => setTimeout(resolve, 5000));
+          // Retry
+          const newRetries = retries - 1;
+          if (newRetries <= 0) {
+            message.error('Failed to create account');
+          } else {
+            message.error('Failed to create account, retrying in 5 seconds');
+          }
+          createSafeWithRetries(retries - 1);
```
Comment (truemiller) [inline frontend/components/SetupPage/Create/SetupCreateSafe.tsx:68]:
```suggestion
          createSafeWithRetries(newRetries);
```

File: frontend/components/SetupPage/Create/SetupCreateSafe.tsx [hunk 3]
```diff
@@ -44,14 +104,18 @@ export const SetupCreateSafe = () => {
...
           <Typography.Text type="secondary" className="mt-12">
-            Error, please contact{' '}
-            <a target="_blank" href={SUPPORT_URL}>
-              Olas community {UNICODE_SYMBOLS.EXTERNAL_LINK}
+            Error, please restart the app and try again.
+          </Typography.Text>
+          <Typography.Text>
+            If the issue persists, please{' '}
+            <a href={SUPPORT_URL} target="_blank" rel="noreferrer">
+              contact Olas community support.
```
Comment (truemiller) [inline frontend/components/SetupPage/Create/SetupCreateSafe.tsx:116]:
```suggestion
              contact Olas community support. {UNICODE_SYMBOLS.EXTERNAL_LINK}
```

**Review by iasonrovis — APPROVED** [review]

**Review by Tanya-atatakai — APPROVED** [review]
agreed with Mohan's comments, but approve in advance

File: frontend/components/SetupPage/Create/SetupCreateSafe.tsx [hunk 4]
```diff
@@ -1,40 +1,101 @@
...
+          // Wait for 5 seconds before retrying
+          await new Promise((resolve) => setTimeout(resolve, 5000));
+          // Retry
+          const newRetries = retries - 1;
+          if (newRetries <= 0) {
+            message.error('Failed to create account');
+          } else {
+            message.error('Failed to create account, retrying in 5 seconds');
+          }
+          createSafeWithRetries(newRetries);
```
[the remark this answers is not in the document]
Comment (truemiller) [reply frontend/components/SetupPage/Create/SetupCreateSafe.tsx:70]: the exit case is at the top of the function which sets states to failed

**Pull request merged**